### DISCUSSION TITLE
Nice level isn't being set

### DISCUSSION THREAD

**Discussion by nonetrix** [fufexan/nix-gaming#190] [Q&A]
I have the following config which should give me nice level -15
```nix
{inputs, ...}: {
  imports = [
    inputs.nix-gaming.nixosModules.pipewireLowLatency
  ];
  security.rtkit.enable = true;
  services.pipewire = {
    enable = true;
    lowLatency.enable = true;
    pulse.enable = true;
    alsa = {
      enable = true;
      support32Bit = true;
    };
  };
}
```
However running `ps -eo pid,comm,nice | grep pipe`  I get the following
```
   2466 pipewire        -11
   2478 pipewire-pulse  -11
```
As you can see it doesn't set to lower than -11, I have been trying to fix my audio crackling issues by setting a lower one to no avail. As a last ditch effort I tried importing this just in case I was doing something wrong, but it still isn't working. This is likely user error or a nix bug or perhaps old configs that might be lingering as I tried to get low latency audio a while back. Right now I am just trying to get the crackle to go away, but since it's happening with this too I guess it wouldn't be a sin to ask here.  I am 80% of the way with getting rid of the crackle, but in extreme cases like maxing out Doom Eternal in a place with a lot of monsters the audio will crackle

**Comment by fufexan** [comment]
Looking at the [upstream pipewire.conf.in](https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/daemon/pipewire.conf.in?ref_type=heads#L117-127), niceness and rt priority levels can be changed from the PipeWire config.

**Comment by nonetrix** [comment]
That's what I did and the nix module I imported does I think?

**Reply by NotAShelf** [reply]
The low latency module as of now is broken after the PW update. I mean to refactor it later.

[Here](https://github.com/NotAShelf/nyx/blob/main/modules%2Fcore%2Fcommon%2Fsystem%2Fhardware%2Fmultimedia%2Fsound%2Fservers%2Fpipewire%2Flow-latency.nix) is a hands-on approach if you want to try it yourself. You will need to tweak the settings to suit your setup.
